### PR TITLE
(WIP) (SIMP-5289) SIMP-6.3.0 updates

### PR DESCRIPTION
**Pull request by lnemsick-simp**
- Use master branches for most simp Puppet modules in
  Puppetfile.tracking, as most versions that will be used in
  SIMP-6.3.0 have not been released
- Add the following dependencies to the simp package
  - pupmod-simp-deferred_resources
  - pupmod-simp-ima
  - pupmod-simp-simp_banners
  - pupmod-simp-tlog
  - pupmod-simp-tpm2
  - pupmod-simp-vox_selinux
- Replace the following dependencies in the simp package
  - pupmod-simp-systemd has been replaced with pupmod-camptocamp-systemd
  - pupmod-simp-timezone has been replaced with pupmod-saz-timezone
- Add the following dependencies to the simp-extras package
  - pupmod-simp-mate
  - pupmod-simp-simp_pki_service
  - pupmod-simp-x2go
- Add tlog application RPM to package.yaml files
- Bump version of herculesteam/augeasproviders_sysctl to
  2.2.1, a version that supports Puppet 5
- Bump version of trlinkin/nsswitch from odd 'simp6.0.0-2.0.0'
  version to official '2.1.0' version

SIMP-4337 #close
SIMP-5289 #close

### CODE REVIEW

**Comment by lnemsick-simp** [comment]
This requires PRs to be merged in order to successfully build an ISO.  simp-core tests have not yet been updated, as some will require updates after all PRs are done (i.e., changes for per-environment hiera.yaml).

File: Puppetfile.tracking [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
-# because camptocamp has moved to Hiera 5 and Puppet >=
-# 4.10.10.  We need to release our version of 1.1.1
-# which is on simp-master.
-mod 'simp-systemd',
+# Revert to camptocamp/systemd for Puppet 5
+# TODO Verify upgrade works switching from simp-systemd to camptocamp-systemd
+mod 'camptocamp-systemd',
   :git => 'https://github.com/simp/puppet-systemd.git',
-  :tag => 'simp-1.1.1'
+  :tag => '2.1.0'
```
Comment (trevor-vaughan) [inline Puppetfile.tracking]: Due to the need for RPM triggers, and the complexity of getting this working in the Oct time frame, I think that we should just update our fork until we can properly fix the module deployment strategy.

File: Puppetfile.tracking [hunk 2]
```diff
@@ -1,59 +1,62 @@
...
 
 mod 'cristifalcas-journald',
   :git => 'https://github.com/simp/pupmod-simp-journald.git',
   :tag => '0.6.0'
 
+#FIXME does not support Puppet 5 yet.  Support added with version 6.0.0.
```
Comment (trevor-vaughan) [inline Puppetfile.tracking]: We should probably just ditch these until we can actually test version 6

File: Puppetfile.tracking [hunk 3]
```diff
@@ -391,124 +417,144 @@ mod 'simp-simp_ipa',
...
+  :branch => 'master'
 
-# SIMP has made changes to this module that don't exist upstream
-# yet. (PR pending). Since may revert to saz/puppet-timezone at a
-# later date, working off of simp-master.
-mod 'simp-timezone',
+# Revert to saz/puppet-timezone for Puppet 5
+# TODO Get RPM triggers to work so that can upgrade from
+#      simp-timezone to saz-timezone
+mod 'saz-timezone',
```
Comment (trevor-vaughan) [inline Puppetfile.tracking]: Due to the need for RPM triggers, and the complexity of getting this working in the Oct time frame, I think that we should just update our fork until we can properly fix the module deployment strategy.

File: metadata.json [hunk 4]
```diff
@@ -311,13 +323,17 @@
       "version_requirement": ">= 6.1.0 < 7.0.0"
     },
     {
-      "name": "simp/timezone",
-      "version_requirement": ">= 4.0.0 < 5.0.0"
+      "name": "saz/timezone",
```
Comment (trevor-vaughan) [inline metadata.json]: Request `simp/timezone` based on messages in `Puppetfile.tracking`

File: src/assets/simp/build/simp.spec [hunk 5]
```diff
@@ -96,18 +99,21 @@ Requires: pupmod-simp-svckill >= 3.2.6, pupmod-simp-svckill < 4.0.0
 Requires: pupmod-simp-swap >= 0.1.1, pupmod-simp-swap < 1.0.0
 Requires: pupmod-simp-tcpwrappers >= 6.0.3, pupmod-simp-tcpwrappers < 7.0.0
 Requires: pupmod-simp-tftpboot >= 6.1.0, pupmod-simp-tftpboot < 7.0.0
-Requires: pupmod-simp-timezone >= 4.0.0, pupmod-simp-timezone < 5.0.0
+Requires: pupmod-saz-timezone >= 5.0.2, pupmod-saz-timezone < 6.0.0
```
Comment (trevor-vaughan) [inline src/assets/simp/build/simp.spec]: Revert to `simp-timezone`

File: Puppetfile.tracking [hunk 6]
```diff
@@ -138,6 +141,8 @@ mod 'puppetlabs-motd',
   :git => 'https://github.com/simp/puppetlabs-motd',
   :tag => '1.9.0'
 
+#FIXME does not support Puppet 5 yet
+# Required by herculesteam-augeasproviders_mounttab
```
Comment (trevor-vaughan) [inline Puppetfile.tracking]: https://tickets.puppetlabs.com/browse/MODULES-8043

**Pull request merged**